tratto/src/main/java/star/tratto/util/javaparser/JavaParserUtils.java

```diff
@@ -0,0 +1,1078 @@
+     *
+     * @param resolvedType a JavaParser resolved type
+     * @return true iff a given type is a type variable of a generic class or
+     * method. If the given type is an array, then this method checks the base
+     * component type.
+     */
+    public static boolean isTypeVariable(
+            ResolvedType resolvedType
+    ) {
+        ResolvedType componentType = removeArray(resolvedType);
+        return componentType.isTypeVariable();
+    }
+
+    /**
+     * Checks if a type name represents a type variable declared in the given
+     * class or method. This method checks both the method where the type is
+     * used and its declaring class.
+     *
+     * @param typeName a type name
+     * @param jpCallable the method using {@code typeName}
+     * @param jpClass the declaring class of {@code jpCallable}
+     * @return true iff a given type is a type variable of the declaring class
+     * or method. If the given type is an array, then this method checks the
+     * base component type.
+     */
+    public static boolean isTypeVariable(
+            String typeName,
+            CallableDeclaration<?> jpCallable,
+            TypeDeclaration<?> jpClass
+    ) {
+        // add all type parameters of the method
+        List<String> typeParameterNames = jpCallable.getTypeParameters()
+                .stream()
+                .map(NodeWithSimpleName::getNameAsString)
+                .collect(Collectors.toList());
+        // add all type parameters of the class
+        if (jpClass instanceof ClassOrInterfaceDeclaration) {
+            typeParameterNames.addAll(
+                    jpClass.asClassOrInterfaceDeclaration().getTypeParameters()
+                            .stream()
+                            .map(NodeWithSimpleName::getNameAsString)
+                            .toList()
+            );
+        }
+        return typeParameterNames.contains(typeName.replaceAll("\\[]", ""));
+    }
+
+    /**
+     * Get all methods available to a given class (including those defined in
+     * superclasses).
+     *
+     * @param typeDeclaration object class
+     * @return list of MethodUsage objects
+     * @throws JPClassNotFoundException if {@code typeDeclaration} is not
+     * resolvable
+     */
+    public static List<MethodUsage> getMethodsOfType(
+            TypeDeclaration<?> typeDeclaration
+    ) throws JPClassNotFoundException {
+        try {
+            return new ArrayList<>(typeDeclaration.resolve().getAllMethods());
+        } catch (UnsolvedSymbolException | IllegalArgumentException e) {
+            String errMsg = String.format(
+                    "Unable to get all the methods of the class %s.%n%s.",
+                    typeDeclaration.getNameAsString(), e
+            );
+            logger.error(errMsg);
+            throw new JPClassNotFoundException(errMsg);
+        }
+    }
+
+    /**
+     * Get all fields available (non-private) to a given class (including
```

Comment:
Why non-private?  Is this intended to be an external client's view of what fields the type has?